test/unit/test-ampdoc.js

```diff
@@ -296,7 +294,16 @@ describes.sandboxed('AmpDocService', {}, () => {
 
       const content2 = document.createElement('span');
       const host2 = document.createElement('div');
-      const shadowRoot2 = host2.createShadowRoot();
+
+      let shadowRoot2;
+      if (isShadowDomSupported()) {
+        if (getShadowDomSupportedVersion() == ShadowDomVersion.V1) {
+          shadowRoot2 = host2.attachShadow({mode: 'open'});
```

Comment:
In this case we can simply do `if (host2.attachShadow) { ...} else {...} `

Reply:
I think that's all that the `getShadowDomSupportedVersion()` function does iirc

Reply:
I have another PR that adds the integration tests for this, and originally I intended to merge this fix with that PR, going to merge this one and we can continue the discussion there if it still needs attention